### ISSUE TITLE
Disable CUDNN download when it's not found at install time

### ISSUE THREAD

**Issue by szagoruyko** [soumith/cudnn.torch#334]
I skimmed through the code and I don't see such option, it tries to download the lib even though it can just be picked up at runtime.

**Comment by borisfom** [comment]
Option may make sense, however: 
What is your exact use-case? Do you have CUDNN installed in location you do not have in your LD_LIBRARY_PATH at the time of install ?

**Comment by szagoruyko** [comment]
I have it in LD_LIBRARY_PATH, it is not picked up. Solved it by local clone and reverting the PR with new cmake file.

**Comment by dasguptar** [comment]
Can this be re-opened?
Even with a local clone of the latest master, it still downloads CUDNN, even though I have CUDNN paths added to my LD_LIBRARY_PATH.

**Comment by borisfom** [comment]
Do you have existing CUDNN 5.1 in your LD_LIBRARY_PATH? It works for me. Also, does it do it more than once for you ?

**Comment by borisfom** [comment]
@soumith: it's trivial to add an option, and I am not even against it. I am just wondering what the use-case is. How do you know before installing Torch for the first time you actually want this option ? 
And once you downloaded it once, it won't do it for this install location anymore - and if the install location is global, then won't ever.

**Comment by dasguptar** [comment]
I have CUDNN version 5.1 in my LD_LIBRARY_PATH. Yet it downloads CUDNN again, both when I do `luarocks install cudnn`, as well as when I do `luarocks make` from a local clone.
Additionally, it also happens when I change my LD_LIBRARY_PATH to point to CUDNN version 5.0 instead of 5.1 as well. 
It does not happen multiple times, since after the first download, the needed files are saved to the `torch/install/include/` and `torch/install/lib/` folders.

**Comment by dasguptar** [comment]
@borisfom @soumith I guess this is fine, as long as a new version of CUDNN does still work on the existing GPU configuration (driver/toolkit/etc.)
But on some older cards, it could be an issue? You guys will know better than me, deferring to your judgement. As of now, the only issue is the mild bother that something existing is being re-downloaded.

**Comment by borisfom** [comment]
@dasguptar Older cards won't be an issue. I have put 5.1 requirement in as there were certain important bug fixes and there are no cases where 5.0 would do better. 
The original extra download of 5.1 should not have happened, if you had cudnn.h in ../include corresponding to where your 5.1 was - because we look for .so.major only, and the minor is only deduced from the header. I think, I probably should change '5.1' back to '5' to accept whatever CUDNN5 is found.

**Comment by soumith** [comment]
i think one central problem is that torch users dont copy over the header (it's not needed for ffi). That's why maybe downloads are happening. If you only check for the correct `.so*` file, then it's better.

**Comment by borisfom** [comment]
Right, I have changed required version to '5' which would fit both cases - with and without headers.
